staging/src/k8s.io/controller-manager/controller/interfaces.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import "net/http"
+
+// Interface defines the base of a controller managed by a controller manager
+type Interface interface {
+	// Name returns the canonical name of the controller.
+	Name() string
+}
+
+// Debuggable defines a controller that requests the controller manager
```

Comment:
Nit, I think it less requests, than it allows.

Reply:
It is indeed the controller manager's decision. Rewording.